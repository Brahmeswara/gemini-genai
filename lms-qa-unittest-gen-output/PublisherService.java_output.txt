package com.example.lws.services;

import com.example.lws.models.Publisher;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.ArgumentMatchers;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.mockito.ArgumentMatchers.anyLong;
import static org.mockito.Mockito.verify;

@ExtendWith(MockitoExtension.class)
class PublisherServiceTest {

    @InjectMocks
    private PublisherService service;

    @Mock
    private PublisherRepository repository;

    @Test
    @DisplayName("Test Save Publisher")
    void savePublisher() {
        Publisher publisher = Publisher.builder().id(1L).name("Publisher 1").build();
        service.savePublisher(publisher);
        verify(repository).save(publisher);
    }

    @Test
    @DisplayName("Test Update Publisher")
    void updatePublisher() {
        Publisher publisher = Publisher.builder().id(1L).name("Publisher 1").build();
        service.updatePublisher(publisher);
        verify(repository).save(publisher);
    }

    @Test
    @DisplayName("Test Delete Publisher")
    void deletePublisher() {
        service.deletePublisher(1L);
        verify(repository).deleteById(1L);
    }

    @Test
    @DisplayName("Test Get Publisher By Id")
    void getPublisherById() {
        service.getPublisherById(1L);
        verify(repository).findById(1L);
    }
}