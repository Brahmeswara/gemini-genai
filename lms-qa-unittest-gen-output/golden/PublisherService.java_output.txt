```java
package com.example.lws.services;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

import com.example.lws.models.Publisher;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.runners.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.class)
public class PublisherServiceImplTest {

  @InjectMocks
  private PublisherServiceImpl publisherService;

  @Mock
  private PublisherRepository publisherRepository;

  private Publisher publisher;

  @Before
  public void setup() {
    publisher = Publisher.builder().id(1L).name("Test Publisher").build();
  }

  @Test
  public void testSavePublisher() {
    Mockito.when(publisherRepository.save(publisher)).thenReturn(publisher);
    Publisher savedPublisher = publisherService.savePublisher(publisher);
    assertNotNull(savedPublisher);
    assertEquals(publisher.getId(), savedPublisher.getId());
  }

  @Test
  public void testUpdatePublisher() {
    Mockito.when(publisherRepository.save(publisher)).thenReturn(publisher);
    Publisher updatedPublisher = publisherService.updatePublisher(publisher);
    assertNotNull(updatedPublisher);
    assertEquals(publisher.getId(), updatedPublisher.getId());
  }

  @Test
  public void testDeletePublisher() {
    publisherService.deletePublisher(publisher.getId());
    Mockito.verify(publisherRepository, Mockito.times(1)).deleteById(publisher.getId());
  }

  @Test
  public void testGetPublisherById() {
    Mockito.when(publisherRepository.findById(publisher.getId())).thenReturn(publisher);
    Publisher foundPublisher = publisherService.getPublisherById(publisher.getId());
    assertNotNull(foundPublisher);
    assertEquals(publisher.getId(), foundPublisher.getId());
  }
}
```